OrbitCore/LinuxUtils.cpp

```diff
@@ -109,6 +109,10 @@ void ListModules(pid_t pid,
     // mapped to a file (might be heap, stack or something else)
     if (tokens.size() != 6 || tokens[4] == "0") continue;
 
+    // tokens[1] is the permissions column. tokens[1][2] is the permission to
+    // be executed. We discard entries that are not executable.
+    if (tokens[1].size() != 4 || tokens[1][2] != 'x') continue;
+
```

Comment:
ok, it appears to be a bit more complicated... if you look into line 121-131, it collects address ranges for maps for every corresponding module. each file usually has 3 entries in /proc/pid/maps the first one is executable, second one is readable and the last one is read-writable. filtering 2 others out will result in some space that is occupied by a service unaccounted.

I think it is better to collect executable flags and after processing everything filtering out files which were not mapped as executables, meaning that all the maps for the file are non-executable.

Reply:
Should be addressed now.